### DISCUSSION TITLE
NextJS 12 + ReactNativeWebView

### DISCUSSION THREAD

**Discussion by interc0der** [vercel/next.js#40144] [Help] [answered]
Is anyone successfully running a NextJS application in a WebView, and can communicate with the webview component using the window.ReactNativeWebView.postMessage()?

I have run a massive amount of testing. 

Every build fails to find the ReactNativeWebView API (ie, window.ReactNativeWebView === 'undefined' ) and I have checked that the window object does exist ( window !== 'undefined')

**Comment by gaston-flores** [comment]
Hey @interc0der! Two questions:

- Are you setting the `onMessage` prop in your WebView? The API won't be injected otherwise.
- Where are you trying to use the `window.ReactNativeWebView` API? Can you share an example?

**Reply by interc0der** [reply]
@gaston-flores 

Example on client side ->
https://github.com/thebettermint/apex-xapp/blob/0e50861a208c372677d977109bdad6d2a38a5c46/src/services/xapp.service.ts#L58

Example on Native Side -> 
https://github.com/XRPL-Labs/XUMM-App/blob/8e6e795648b97d0067380baa43bedea9e4b5ea61/src/screens/Modal/XAppBrowser/XAppBrowserModal.tsx#L515

**Reply by interc0der** [reply]
@gaston-flores
I am calling the function in a few locations. Here is an example on a page -> https://github.com/thebettermint/apex-xapp/blob/0e50861a208c372677d977109bdad6d2a38a5c46/src/pages/share.tsx#L90

Auto deploying on vercel

**Comment by WietseWind** [comment, marked as answer]
@interc0der I'm terribly sorry, but I believe you are encountering this because you are building a [Xumm xApp](https://xumm.readme.io/docs/xapp-xumm-ui-interaction#qr-code-scanner-opened--scanned--closed)?

If so: please note we no longer expose the `window.ReactNativeWebView` in xApps: the only (and improved, cleaner) way to interact with the Xumm host application is through our SDK.

So it can very well be you're not finding the object because it really isn't there. This may be on us, instead of `next.js` (I just tested it, it just works in next.js, if it wasn't for our new xApp SDK version.

Terribly sorry for the wasted time and confusion.
